src/loaders/mod_load.c

```diff
@@ -602,21 +641,27 @@ static int mod_load(struct module_data *m, HIO_HANDLE *f, const int start)
     }
 
 skip_test:
+#endif
 
     if (mod->chn >= XMP_MAX_CHANNELS) {
         return -1;
     }
 
     mod->trk = mod->chn * mod->pat;
 
+    #ifndef LIBXMP_CORE_PLAYER
     /* Digital Tracker MODs have an extra four bytes after the magic.
      * These are always 00h 40h 00h 00h and can probably be ignored. */
     if (tracker_id == TRACKER_DIGITALTRACKER) {
 	hio_read32b(f);
     }
+    #else
+    libxmp_set_type(m, (mod->chn == 4) ? "Protracker" : "Fasttracker");
+    MODULE_INFO();
+    #endif
 
     for (i = 0; i < mod->ins; i++) {
-	D_(D_INFO "[%2X] %-22.22s %04x %04x %04x %c V%02x %+d %c",
+	D_(D_INFO "[%2X] %-22.22s %04x %04x %04x %c V%02x %+d %c\n",
```

Comment:
`D_()` already injects a `\n`, I think this line is safe to revert.

Reply:
Ehh, forgot removing this one, sorry. Can you send a patch packed with the hio_read buffer thing below for this?